client.go

```diff
@@ -41,48 +41,56 @@ type SingleServerClient struct {
 	isLocalTest bool
 }
 
-func NewClientImpl(deployFile string, targetName string, isLocalTest bool) (*SingleServerClient, error) {
+// Client which communicates with multiple underlying servers at once. Used if
+// the target is actually a group of servers
+type MultiServerClient struct {
+	app     Application
+	appDir  string
+	clients []Client
+}
+
+func NewClient(deployFile string, targetName TargetName, isLocalTest bool) (*MultiServerClient, error) {
 	app, err := ApplicationFromConfig(true, deployFile)
 	if err != nil {
 		return nil, err
 	}
 
-	target := app.Target(targetName)
-	if target == nil {
-		return nil, fmt.Errorf("Invalid target: '%s'", targetName)
-	}
+	// Create all SingleServerClients
+	clients := []Client{}
+	for _, target := range app.Targets(targetName) {
+		localPort := target.Base
```

Comment:
I think there should be a different local port for each client (won't this not work with more than one client?)

In fact, why does it even need to be the same as the Base (this is probably why we have port conflicts in the first place - presumably a leftover from a very early version of camus)  - assuming I still remember correctly what these fields are for.